lib/Controller/ConfigController.php

```
@@ -364,4 +379,33 @@ public function checkConfig(): DataResponse {
 			'authorization_header' => $_SERVER['HTTP_AUTHORIZATION'],
 		]);
 	}
+
+	/**
+	 * Revokes the personal OAuth access token for the provided user using the current client information
+	 * Token should be the one provided by the OP OAuth client and stored in the NC database
+	 *
+	 * @param string $userId - user whose OAuth access token is to be revoked
+	 * @param string $opOAuthClientID - OAuth client from where token is to be revoked
+	 * @param string $opOAuthClientSecret - OAuth client secret from where token is to be revoked
+	 *
+	 * @return void
+	 * @throws OpenprojectErrorException
+	 */
+	private function revokeUserToken(
+		string $opUrl,
+		string $userId,
+		string $opOAuthClientID,
+		string $opOAuthClientSecret
+	): void {
+		$accessToken = $this->config->getUserValue($userId, Application::APP_ID, 'token', '');
+
+		if ($opUrl && $opOAuthClientID && $opOAuthClientSecret && $accessToken) {
```

what is that expression is not true, then the code will silently do nothing. I think its better to throw an exception

---

there can be case where user might not have the `accessToken` set. I've modified the code. Please review.